projects/js-packages/publicize-components/src/hooks/use-attached-media/index.js

```diff
@@ -0,0 +1,30 @@
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+
+/**
+ * @typedef {Object} AttachedMediaHook
+ * @property {Array} attachedMedia - List of media IDs.
+ * @property {Function} updateAttachedMedia - Callback used to update the attached media..
+ */
+
+/**
+ * Hook to handle storing the attached media.
+ *
+ * @returns {AttachedMediaHook} - An object with the attached media hook properties set.
+ */
+export default function useAttachedMedia() {
+	const { editPost } = useDispatch( editorStore );
+
+	const attachedMedia = useSelect( select => select( 'jetpack/publicize' ).getAttachedMedia() );
+
+	return {
+		attachedMedia,
+		updateAttachedMedia: function ( ids ) {
+			editPost( {
+				meta: {
+					jetpack_publicize_attached_media: ids,
+				},
+			} );
+		},
```

Comment:
I wonder if this should be wrapped in `useCallback`? I don't know for sure the performance implications, but `updateAttachedMedia` will change on each call to the hook, and that will invalidate the `useCallback` calls, that have it in its dependencies.

Maybe we could implement dependencies for this hook?

Reply:
Can you elaborate a bit on this? I think it makes sense in theory, but I'm not sure exactly where to add the `useCallback`. Do you mean wrapping the `editPost` in a `useCallback`?